vispy/visuals/tests/test_axis.py

```diff
@@ -64,4 +64,57 @@ def test_rotation_angle():
     assert_allclose(axis2._rotation_angle, -17.056795, rtol=0.05)
 
 
+@requires_application()
+def test_text_position():
+
+    # Test the labels text and position of the axis depending on its domain
+
+    canvas = scene.SceneCanvas(keys=None, size=(800, 600), show=True)
+    view = canvas.central_widget.add_view()
+    view.camera = scene.cameras.PanZoomCamera(parent=view.scene)
+
+    # tick length and label margin to 0 for nice values
+    axis1 = visuals.Axis(pos=[[-1.0, 0], [1.0, 0]], domain=(0., 1.25), 
+                         major_tick_length=0, tick_label_margin=0, 
+                         parent=view.scene)
+
+    canvas.draw_visual(axis1)
+    assert_allclose(axis1._text.pos[:, 0], (-1, -0.2, 0.6))
+    assert_array_equal(axis1._text.text, ('0', '0.5', '1'))
+
+    # Flip the axis domain
+    axis1.domain = (1.25, 0.)
+    canvas.draw_visual(axis1)
+    # Text should be unchanged and positions mirrored
+    assert_allclose(axis1._text.pos[:, 0], (1, 0.2, -0.6))
+    assert_array_equal(axis1._text.text, ('0', '0.5', '1'))
+
+
+@requires_application()
+def test_tick_position():
+
+    # Test the position of the ticks depending on the axis domain
+
+    canvas = scene.SceneCanvas(keys=None, size=(800, 600), show=True)
+    view = canvas.central_widget.add_view()
+    view.camera = scene.cameras.PanZoomCamera(parent=view.scene)
+
+    axis1 = visuals.Axis(pos=[[-1.0, 0], [1.0, 0]], domain=(0., 1.25), parent=view.scene)
+
+    canvas.draw_visual(axis1)
+    # Get a nice array of x ticks positions
+    x_ticks_positions = axis1._ticks.pos[:, ::2][::2].flatten()
```

Comment:
Why do you take every other (`::2`) point?

Reply:
Ticks positions are given for each tick line, so each tick has two (x, y) positions. As the axis is x-oriented, for each line, both x positions are equals so we can remove half of them.

As we are not interested in y positions either (again, we are testing an x-oriented axis), we can again remove half of the points.

Reply:
Got it. Sounds good. Seeing as your comment is from 8 hours ago, do you plan on making that commit with the change?